README.md

```diff
@@ -26,9 +25,54 @@
     - Test the program in different maze
 
 
-
-
-
+# Program structure (Pseudo Code) - might be a little different from real code in terms of variable names
+ main 
+ #define variables 
+ [$s0]: address of maze array
+ [$s1]: mazewidth 
+ [$t0]: starting-xcord 
+ [$t1]: starting-ycord
+ [$t2]: number of mistakes
+ [$t3]: number of moves
+ **to do: maze selection could be placed here**
+ > j {main_loop}
+	> jal {get_user_input}
+		- # print welcome
+		- # call for user input, save to [$v1]
+		- beq: # check if valid:
```

Comment:
validity checks happen uniquely for each movement direction (check_left, check_forward, etc)